enhancements/microshift/optionally-deploy-csi-plugin.md

```diff
@@ -0,0 +1,256 @@
+6. User deletes the relevant cluster API resources
+    ```shell
+    $ oc delete -n kube-system deployment.apps/csi-snapshot-controller deployment.apps/csi-snapshot-webhook   
+    $ oc delete -n openshift-storage daemonset.apps/topolvm-node
+    $ oc delete -n openshift-storage deployment.apps/topolvm-controller
+    $ oc delete -n openshift-storage configmaps/lvmd
+    $ oc delete oc get storageclasses.storage.k8s.io/topolvm-provisioner
+    ```
+7. The cluster processes the deletions and the process is complete
+
+### API Extensions
+
+- None
+
+### Topology Considerations
+
+#### Single-node Deployments or MicroShift
+
+The changes proposed here only affect MicroShift.
+
+### Implementation Details/Notes/Constraints
+
+- **Rebase Changes:** 
+  - LVMS and the snapshotting manifests will be moved from `microshift/assets/components/lvms` and 
+  `microshift/assets/components/csi-snapshot-controller` to `microshift/assets/optional/lvms` and 
+  `microshift/assets/optional/csi-snapshot-controller`. These paths will be updated in the 
+    `microshift/scripts/auto-rebase/lvms_assets.yaml` to reflect the new paths.
+  - LVMS image digests will be moved from `microshift/assets/release/release-$ARCH.json` to 
+    `microshift/assets/optional/lvms/release-$ARCH.json`.
+  - CSI Snapshotter image digests will be moved from `microshift/assets/release/release-$ARCH.json` to 
+      `microshift/assets/optional/csi-snapshot-controller/release-$ARCH.json`.
+  - During rebasing, the image digests for LVMS and CSI Snapshotter will be parsed from the LVMS bundle (like they 
+    are now) and the digests will be written to their respective `release-$ARCH.json` files.
+
+- **Mainline Code Changes:**
+  - MicroShift source code will no longer manage LVMS or CSI manifest deployment or its configuration. The 
+    microshift service manager which handles these components will be removed entirely. The following files will be 
+    deleted:
+    - `microshift/pkg/components/csi-snapshot-controller.go`
+    - `microshift/pkg/components/storage.go`
+    - `microshift/pkg/components/render_test.go`, which only tests LVMS parameter rendering
+    - `microshift/pkg/assets/storage.go`
+  - Additionally, certain functions related to managing LVMS will be deleted:
+    - `microshift/pkg/components/render_test.go:startCSIPlugin()`
+    - `microshift/pkg/components/render_test.go:startCSISnapshotterController()`
+  - MicroShift also contains logic to determine if LVM is installed and attempt a best-guess at which volume group, 
+    if any, should be set in the LVMD config. This code will need to be removed and implemented as:
+    - Specifying LVM as an RPM dependency of `microshift-lvms`.
+    - A `microshift-lvms` rpm post-install script that determines what volume group, if any, should be specified.
+
+- **RPM Spec**
+  - New RPM specs will be written for `microshift-lvms` and `microshift-lvms-snapshotting` packages. These have the 
```

Comment:
Since snapshotting can be used for any CSI driver, WDYT about microshift-snapshotting?

That way anyone who is just interested in snapshotting capability can install that RPM and then install their own CSI driver + VolumeSnapshotClass.

LVMS can already smartly create or ignore VolumeSnapshotClass creation so this would not need any extra logic paths.

Reply:
Addressed in an earlier comment. LVMS and Snapshotting are considered as separate installs in the EP